### PR TITLE
feat(widget-builder): implement responsive preview

### PR DESCRIPTION
**Pull request by nikkikapadia**
i'm back again with the right implementation this time LOLL. When the slideout is opened on a smaller screen, the preview is embedded in the slideout initially until you scroll past it. Once you scroll past it, it pops up as a floating, draggable preview. This dissapears when the embedded preview is in view again. 

There are some things that are left to implement:
1. Some aesthetic fixes for big number widget previews
2. Design accurate draggable preview
3. Snapping the draggable preview into one of four corners

This is what it looks like right now:

https://github.com/user-attachments/assets/310bb970-dfc1-4a0e-aea1-a8b639777029

### CODE REVIEW

File: static/app/views/dashboards/widgetBuilder/components/newWidgetBuilder.tsx [hunk 1]
```diff
@@ -48,6 +51,29 @@ function WidgetBuilderV2({
...
+  useEffect(() => {
+    const handleResize = () => {
+      setWindowWidth(window.innerWidth);
+    };
+
+    window.addEventListener('resize', handleResize);
+    return () => window.removeEventListener('resize', handleResize);
+  }, []);
+
+  const isSmallScreen =
```
Comment (narsaynorath) [inline static/app/views/dashboards/widgetBuilder/components/newWidgetBuilder.tsx:67]: It looks like there's a hook for this called `useMedia`, if you check the code base for its usage we could probably use that?

Comment (narsaynorath) [reply static/app/views/dashboards/widgetBuilder/components/newWidgetBuilder.tsx:67]: I think that also means we can get rid of the `useEffect` here, right? Let's see if that could be removed in the other file too

**Review by narsaynorath — APPROVED** [review]
Just a small comment on how we're calculating `isSmallScreen`, but this looks great!

File: static/app/views/dashboards/widgetBuilder/components/newWidgetBuilder.tsx (hunk 1 above)
Comment (nikkikapadia) [reply static/app/views/dashboards/widgetBuilder/components/newWidgetBuilder.tsx:67]: oh no wayyy, that's super helpful

**Pull request merged**